includes/wikia/tasks/AsyncTaskList.class.php

```diff
@@ -284,12 +285,14 @@ protected function getExecutor() {
 	 * @throws AMQPExceptionInterface
 	 */
 	public function queue( AMQPChannel $channel = null, $priority = null ) {
-		global $wgUser;
-
 		$this->initializeWorkId();
 
 		if ( $this->createdBy == null ) {
-			$this->createdBy( $wgUser );
+			global $wgUser, $wgRequest;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)